canal/canal.go

```
@@ -199,28 +204,62 @@ func (c *Canal) GetTable(db string, table string) (*schema.Table, error) {
 		return t, nil
 	}
 
+	c.tableLock.RLock()
+	lastTime, ok := c.errorTablesGetTime[key]
+	c.tableLock.RUnlock()
+
+	if ok && time.Now().Sub(lastTime) < UnknowTableRetryIntervalSec {
+		return nil, schema.LastGetTableInfoErr
+	}
+
 	t, err := schema.NewTable(c, db, table)
 	if err != nil {
 		// check table not exists
 		if ok, err1 := schema.IsTableExist(c, db, table); err1 == nil && !ok {
 			return nil, schema.ErrTableNotExist
 		}
-
-		return nil, errors.Trace(err)
+		// work around : RDSHeartBeat
+		// https://github.com/alibaba/canal/blob/master/parse/src/main/java/com/alibaba/otter/canal/parse/inbound/mysql/dbsync/LogEventConvert.java#L385
+		if key == schema.HAHealthCheckSchema {
+			// mock ha_health_check meta
+			ta := &schema.Table{
+				Schema:  db,
+				Name:    table,
+				Columns: make([]schema.TableColumn, 0, 2),
+				Indexes: make([]*schema.Index, 0),
+			}
+			ta.AddColumn("id", "bigint(20)", "", "")
+			ta.AddColumn("type", "char(1)", "", "")
+			c.tableLock.Lock()
+			c.tables[key] = ta
+			c.tableLock.Unlock()
+			return ta, nil
+		}
+		c.tableLock.Lock()
+		c.errorTablesGetTime[key] = time.Now()
+		c.tableLock.Unlock()
+		// log error and return LastGetTableInfoErr
+		log.Errorf("GetTable info error %v", errors.Trace(err))
+		return nil, schema.LastGetTableInfoErr
```

bad to ignore any error here, please just handle the special case here.